backends-velox/src/test/scala/io/glutenproject/execution/VeloxDataTypeValidationSuite.scala

```diff
@@ -234,4 +234,23 @@ class VeloxDataTypeValidationSuite extends WholeStageTransformerSuite {
     runQueryAndCompare("select type1.array from type1," +
       " type2 where type1.array = type2.array") { _ => }
   }
+
+  test("Map type") {
+    // Validation: BatchScan Project Limit
+    runQueryAndCompare("select map from type1 limit 1") { _ => }
+    // Validation: BatchScan Project Aggregate Sort Limit
+    // TODO validate Expand operator support map type ?
+    runQueryAndCompare("select map['key'] from type1 group by map['key']" +
+      " sort by map['key'] limit 1") { _ => }
+
+    // Validation: BroadHashJoin, Filter, Project
+    super.sparkConf.set("spark.sql.autoBroadcastJoinThreshold", "10M")
+    runQueryAndCompare("select type1.map['key'] from type1," +
+      " type2 where type1.map['key'] = type2.map['key']") { _ => }
+
+    // Validation: ShuffledHashJoin, Filter, Project
+    super.sparkConf.set("spark.sql.autoBroadcastJoinThreshold", "-1")
+    runQueryAndCompare("select type1.map['key'] from type1," +
+      " type2 where type1.map['key'] = type2.map['key']") { _ => }
```

Comment:
I see this PR added some support for map type in both cpp code and ArrowWritableColumnVector, will these code be executed in above unit test? Actually I'm not clear whether this test is executed by native or fallbacked into vanilla spark.

Reply:
@rui-mo 
In this test, the Project, Limit and Filter Node will be offload to native and other operators will fallback to vanilla spark. So the ArrowWritableColumnVector code will be executed.

Here is one of the physical plan:
`
== Physical Plan ==

CollectLimit 1
+- *(1) ColumnarToRow
   +- VeloxLoadArrowData
      +- *(2) ProjectExecTransformer [map#14]
         +- VeloxRowToArrowColumnar
            +- BatchScan[map#14] ParquetScan DataFilters: [], Format: parquet, Location: InMemoryFileIndex(1 paths)[file:/home/sparkuser/jk/projects/gluten/backends-velox/target/scala-2...., PartitionFilters: [], PushedFilters: [], ReadSchema: struct<map:map<string,string>>, PushedFilters: [] RuntimeFilters: []
`

Reply:
Thanks for the explain!